### PR TITLE
Reddit Graph dataset

### PR DESCRIPTION
**Pull request by Dsantra92**
Added the Reddit dataset from [Inductive Representation Learning on Large Graphs](https://arxiv.org/abs/1706.02216).

### CODE REVIEW

File: src/datasets_graph/Reddit/Reddit.jl [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+# References
+[1]: [Inductive Representation Learning on Large Graphs](https://arxiv.org/abs/1706.02216)
+[2]: [Benchmarks on the Reddit Dataset](https://paperswithcode.com/dataset/reddit)
+"""
+module Reddit
+using DataDeps
+using JSON3
+using ..MLDatasets: datafile
+using NPZ: npzread
```
[the remark this answers is not in the document]
Comment (Dsantra92) [reply src/datasets_graph/Reddit/Reddit.jl:26]: `Pickle.npyload` throws errror with the features `.npy` file
```
ERROR: ArgumentError: Deque must be non-empty
Stacktrace:
  [1] pop!
    @ ~/.julia/packages/DataStructures/vSp4s/src/deque.jl:274 [inlined]
  [2] pop!
    @ ~/.julia/packages/DataStructures/vSp4s/src/stack.jl:51 [inlined]
  [3] pop!
    @ ~/.julia/packages/Pickle/iHRCQ/src/Pickle.jl:44 [inlined]
  [4] execute!(p::Pickler{4}, #unused#::Val{Pickle.OpCodes.STACK_GLOBAL}, arg::Nothing)
    @ Pickle ~/.julia/packages/Pickle/iHRCQ/src/deserializer.jl:163
  [5] run!(p::Pickler{4}, op::Pickle.OpCodes.OpCode, io::IOStream)
    @ Pickle ~/.julia/packages/Pickle/iHRCQ/src/deserializer.jl:25
  [6] load(p::Pickler{4}, io::IOStream)
    @ Pickle ~/.julia/packages/Pickle/iHRCQ/src/deserializer.jl:15
  [7] #15
    @ ~/.julia/packages/Pickle/iHRCQ/src/deserializer.jl:11 [inlined]
  [8] open(f::Pickle.var"#15#16"{Pickler{4}}, args::String; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./io.jl:330
  [9] open(f::Function, args::String)
    @ Base ./io.jl:328
 [10] load
    @ ~/.julia/packages/Pickle/iHRCQ/src/deserializer.jl:11 [inlined]
 [11] npyload(f::String)
    @ Pickle ~/.julia/packages/Pickle/iHRCQ/src/np.jl:24
 [12] top-level scope
    @ REPL[6]:1
```
I looked around this error, it has something to do with `Pickle` itself. [This StackOverflow](https://stackoverflow.com/a/50472787/12902351) points out some issues that indicate the problem might be how the `.npy` was serialized.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
[the remark this answers is not in the document]
Comment (Dsantra92) [reply .vscode/settings.json:2]: That slipped out. Adding it to .gitignore.

File: src/datasets_graph/Reddit/Reddit.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuehhua) [reply src/datasets_graph/Reddit/Reddit.jl:26]: It works here: https://github.com/yuehhua/GraphMLDatasets.jl/blob/9badb3bd44d1d8d19a140b5e78365a97a7bf6f0b/src/preprocess.jl#L134
It needs some tricks.

[the remark this answers is not in the document]
Comment (CarloLucibello) [reply src/datasets_graph/Reddit/Reddit.jl:26]: @Dsantra92 can you try that approach?

[the remark this answers is not in the document]
Comment (Dsantra92) [reply src/datasets_graph/Reddit/Reddit.jl:26]: Yes, this looks like a better approach, and I have started working on it. I am busy with some uni work today, committing the changes by tomorrow.

[the remark this answers is not in the document]
Comment (Dsantra92) [reply src/datasets_graph/Reddit/Reddit.jl:26]: The method suggested by @yuehhua leads uses NPZ at it's core.

`Pickle.jl` is sparsely documented but according to [a comment](https://github.com/chengchingwen/Pickle.jl/issues/17#issuecomment-909402006) by the maintainer `Pickle.npyload` is for loading pickle files. The `.npy` file we have at hand is not a pickle (I tested with the latest version of python and numpy), so I believe, we cannot use `Pickle.npyload` here. Referencing my previous comment, the `.npy` was not serialized as a pickle, so it won't load as a pickle.

**Pull request merged**